.github/pull_request_template.md

```diff
@@ -8,13 +8,14 @@ If some of the following don't apply, delete the relevant line.
   See [Changes files](https://github.com/fleetdm/fleet/blob/main/docs/Contributing/Committing-Changes.md#changes-files) for more information.
 - [ ] Input data is properly validated, `SELECT *` is avoided, SQL injection is prevented (using placeholders for values in statements)
 - [ ] Added support on fleet's osquery simulator `cmd/osquery-perf` for new osquery data ingestion features.
-- [ ] Added/updated tests
 - [ ] If paths of existing endpoints are modified without backwards compatibility, checked the frontend/CLI for any necessary changes
 - [ ] If database migrations are included, checked table schema to confirm autoupdate
 - For database migrations:
   - [ ] Checked schema for all modified table for columns that will auto-update timestamps during migration.
   - [ ] Confirmed that updating the timestamps is acceptable, and will not cause unwanted side effects.
   - [ ] Ensured the correct collation is explicitly set for character columns (`COLLATE utf8mb4_unicode_ci`).
+- [ ] Added/updated automated tests
+- [ ] A detailed QA plan exists on the associated ticket (if it isn't there, add it)
```

Comment:
```suggestion
- [ ] A detailed QA plan exists on the associated ticket (if it isn't there, work with the product group's QA engineer add it)
```

Reply:
"to add it" ?

Reply:
Test plans will be official QA responsibilities, so I'm adding this note to work with QA Engineer to make it happen. 

The goal is for test plans to be written before the story ever reaches estimation so that QA, design, and engineering can all align on the "definition of done". The problem with the engineer writing it is that if the engineering misunderstands the spec, it gets codified into the test plan.

Reply:
But this is a great start because definitely we should have test plans before it goes to ready for QA.

Reply:
Yeah. This can be an interim measure for stories. Intent here is to avoid QA plans getting dropped on the floor until after merged, which is more likely on bugs since we're intentionally light on what's required in the initial ticket.

More thoughts on this once I'm out of retro :)

Reply:
So, across the board QA should be the ones signing off on test plans from an "is this thorough enough" perspective. Question is:

1. When does that back-and-forth ideally happen for stories (which you answered above)
2. When does that back-and-forth ideally happen for bugs, given that we don't even have an commented-out/optional section for "when fixed this should work this way" (should we?)

Right now in practice this responsibility is shifted well to the right of where it should be, with test plan questions coming up after PRs have been marged into `main`, and I think those questions tend to be QA asking eng, particularly on bugs (since bugs don't necessarily go through product).

Which brings up the question of, if a test plan got missed, should a PR even be considered review-ready prior to a test plan getting there? Is a test plan, signed off by QA, as much of a merge blocker as e.g. a changes file or eng self-QA is?

I think that ideally the checkbox is a no-op, but as long as we have non-trivial bugfixes where desired behavior and regression test/non-happy-path scenarios aren't specified by the bug reporter it'll come in handy.

Either way, agreed with the revision (net of typo fix) so fixing that now.